crates/oxc_linter/src/rules/unicorn/no_abusive_eslint_disable.rs

```diff
@@ -0,0 +1,139 @@
+use lazy_static::lazy_static;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use regex::Regex;
+
+use crate::{context::LintContext, disable_directives::DisableRuleComment, rule::Rule};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-unicorn(no-abusive-eslint-disable): Unexpected `eslint-disable` comment that does not specify any rules to disable.")]
+#[diagnostic(severity(warning), help("Specify the rules you want to disable."))]
+struct NoAbusiveEslintDisableDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoAbusiveEslintDisable;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// This rule disallows `eslint-disable` comments that do not specify any rules to disable.
+    ///
+    /// ### Why is this bad?
+    /// When only one rule should be disabled but the `eslint-disable` comment does not specify any rules, other useful errors will also be silently ignored.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Fail
+    /// /* eslint-disable */
+    /// console.log(message);
+    ///
+    /// console.log(message); // eslint-disable-line
+    ///
+    /// // eslint-disable-next-line
+    /// console.log(message);
+    ///
+    /// // Pass
+    /// /* eslint-disable no-console */
+    /// console.log(message);
+    ///
+    /// console.log(message); // eslint-disable-line no-console
+    ///
+    /// // eslint-disable-next-line no-console
+    /// console.log(message);
+    /// ```
+    NoAbusiveEslintDisable,
+    restriction
+);
+
+impl Rule for NoAbusiveEslintDisable {
+    fn run_once(&self, ctx: &LintContext) {
+        lazy_static! {
+            static ref RULE_PATTERN: Regex =
+                Regex::new("^(?:@[0-9A-Za-z_-]+/)?(?:[0-9A-Za-z_-]+/)?[0-9A-Za-z_-]+$").unwrap();
```

Comment:
It seems like this can be simplified to a simple .starts('@') + split('/')? Would you like to give it a try?

Reply:
Sure. Done in 398456c19cd506314b141861d82f54d0b3385560.